### PR TITLE
use tensor dtype for zeros_like()

### PR DESCRIPTION
**Pull request by ubihot**
- [x] need to add tests

### CODE REVIEW

File: tinygrad/tensor.py [hunk 1]
```diff
@@ -128,7 +128,7 @@ def zeros(*shape, **kwargs): return Tensor([0], **kwargs).reshape([1]*len(shape)
   def ones(*shape, **kwargs): return Tensor([1], **kwargs).reshape([1]*len(shape)).expand(shape).contiguous()
 
   @staticmethod
-  def zeros_like(tensor, **kwargs): return Tensor.zeros(*tensor.shape, **kwargs)
+  def zeros_like(tensor, **kwargs): return Tensor.zeros(*tensor.shape, dtype = tensor.dtype, **kwargs)
```
Comment (python273) [inline tinygrad/tensor.py:131]: remove spaces `_=_` -> `=`

File: test/test_tensor.py [hunk 2]
```diff
@@ -154,5 +155,34 @@ def test_random_fns_are_deterministic_with_seed(self):
...
         np.testing.assert_allclose(a.numpy(), b.numpy())
 
+  def test_zeros_like_has_same_dtype(self):
+    a = Tensor([1, 2, 3])
+    b = Tensor.zeros_like(a)
+    self.assertTrue(a.dtype == b.dtype)
+
+    a = Tensor([1, 2, 3], dtype=dtypes.float16)
+    b = Tensor.zeros_like(a)
+    self.assertTrue(a.dtype == b.dtype)
```
Comment (python273) [inline test/test_tensor.py:165]: I'd say just 2 cases is enough (default and say int8), you can add `assert` to check shapes are the same

File: test/test_tensor.py [hunk 3]
```diff
@@ -154,5 +155,11 @@ def test_random_fns_are_deterministic_with_seed(self):
         b = random_fn(10,10).realize()
         np.testing.assert_allclose(a.numpy(), b.numpy())
 
+  def test_zeros_like_has_same_dtype(self):
+    for dtype in dtypes.__annotations__:
+      a = Tensor([1, 2, 3], dtype=dtypes.float16)
+      b = Tensor.zeros_like(a)
+      self.assertTrue(a.dtype == b.dtype)
```
Comment (python273) [inline test/test_tensor.py:162]: remove loop, don't really need to test every dtype (and currently you have a bug, `dtype` var is unused)

just have 2 cases and add shape assert

File: test/test_tensor.py (hunk 2 above)
Comment (ubihot) [reply test/test_tensor.py]: @python273 updated the test, let me know if there's a better way of doing this

**Comment by geohot** [comment]
Thanks for the fix.

**Pull request merged**